cobalt/black_box_tests/black_box_tests.py

```diff
@@ -213,50 +219,49 @@ def LoadEvergreenEndToEndTests(launcher_params):
 class BlackBoxTests(object):
   """Helper class to run all black box tests and return results."""
 
-  def __init__(self,
-               server_binding_address,
-               proxy_address=None,
-               proxy_port=None,
-               test_name=None,
-               wpt_http_port=None,
-               device_ips=None,
-               device_id=None):
+  def __init__(self, args):
+
+    self.args = args
+
+    #TODO(b/137905502): These globals should be refactored
     # Setup global variables used by test cases.
     global _launcher_params
     _launcher_params = command_line.CreateLauncherParams()
     # Keep other modules from seeing these args.
     sys.argv = sys.argv[:1]
-    global _binding_address
-    _binding_address = server_binding_address
+    global _server_binding_address
+    _server_binding_address = args.server_binding_address
+
     # Port used to create the web platform test http server. If not specified,
     # a random free port is used.
-    if wpt_http_port is None:
-      wpt_http_port = str(self.GetUnusedPort([server_binding_address]))
     global _wpt_http_port
-    _wpt_http_port = wpt_http_port
+    _wpt_http_port = args.wpt_http_port or str(
+        self.GetUnusedPort([_server_binding_address]))
+
+    # Proxy is only needed for WPT
+    self.use_proxy = args.test_set in ['all', 'wpt']
```

Comment:
It looks like when `--test_set` is "all" (the default) the proxy will be used for all tests, not just the "wpt" ones.

Is that intended / is there a good way to remove the proxy when the evergreen and blackbox tests are run as part of an "all" run? Or will CI make separate calls to black_box_tests.py for each type of test instead of using the "all" option?

Reply:
I'm planning to go through stepwise breaking up here. I want to get a change in here first where "all" is the default behavior that doesn't change how anything is run. That keeps internal CI and at least initially, on-device tests running the same way.

We'll go step by step and switch callers ( Actions, on-device test lab ) to be explicit about which tests they are running once this is in. Sample changes for Linux+Evergreen-x86 on host tests are in the second commit here.

Also, i don't see a lot of reason to bundle all of this complexity into a single file, i think it'll end up looking cleaner if the entry point `.py` file to WPT and blackboxtest is going to be separate, using shared module code as appropriate.

Reply:
Ok cool, that all sounds good. LGTM.